src/home/home.service.ts

```diff
@@ -79,4 +81,16 @@ export class HomeService {
       interests: interests,
     };
   }
+
+  async globalSearch(pagination: PaginationDto, query: HomeQuery) {
+    const [event, group] = await Promise.all([
+      this.eventService.searchAllEvents(pagination, query),
```

Comment:
Well, I figured out now that you are pulling all nested entities on a simple global search request. All needed on frontend is event/group name and it's slug for now.